### DISCUSSION TITLE
error WIX9100: HeatWave Build Tools license failure.

### DISCUSSION THREAD

**Discussion by ArmorBearer** [wixtoolset/issues#8717] [Questions]
I tried to create a WiX v4 demo, following the steps in the tutorial.  So far, so good.

Then I tried modifying the demo to use Heat to harvest the files, rather than specifying them manually.  Not so good.

When I installed the extension for Heat (FireGiant.HeatWave.BuildTools), it seemed to be only available for v.5.0.1.  So I tried to update my demo for WiX v5.  Since the tutorial for v5 appears to be the same, I simply renamed a few files to indicate that it was a v5 demo.

I also installed the Nuget package for WixToolset.Util v.5.0.1.

When I rebuild the project, I get this error message:

> error WIX9100: HeatWave Build Tools license failure. No license found. Checked for licenses at the standard folder or below the directory: 'C:\Projects\WixV5Tutorial\WixV5TutorialPackage'. For more information see: https://www.firegiant.com/wix/wep-documentation/getting-started/firegiant-licensing/

When I go to the web page referenced by the 'more information' link, I find no information about this error - only a description of a license file, which is apparently needed.  A statement helpfully explains:
> You can download your license file from [your FireGiant Account's License page](https://account.firegiant.com/licenses/).

I'm not aware that I have an account.  When I click on this link to go to the licenses page, I'm instructed to login.  The friendly message says,
> New here? Enter your email address and we'll get you set up.

So I enter my email address, and I'm asked for a new password, as well as a confirmation.  After entering these, I'm told my password is invalid, with no further explanation.

What's the secret sauce for getting Heat to work?

**Comment by robmen** [comment]
Hmm, a couple of things went awry here.

First, the login problem is troubling. Could you try [refreshing the login page](https://account.firegiant.com/) and then trying logging in? If that doesn't work, I'd love for our support team to fix it for you, so send an email to `support at firegiant.com` with the email address that is having the problems, and they'll dig into it.

Second, `FireGiant.HeatWave.BuildTools` NuGet package contains the [Advanced Harvesting feature](https://www.firegiant.com/docs/heatwave-build-tools/harvesting/) (among other features). If you want the old `Heat.exe` then you want the `WixToolset.Heat` NuGet package.

**Reply by ArmorBearer** [reply]
> FireGiant.HeatWave.BuildTools NuGet package contains the [Advanced Harvesting feature](https://www.firegiant.com/docs/heatwave-build-tools/harvesting/) (among other features). If you want the old Heat.exe then you want the WixToolset.Heat NuGet package.

Which one do I need in order to use Heat with WiX v.5?  There seems to be little documentation about how to use harvesting in WiX v5, other than the Advanced Harvesting page...

**Reply by ArmorBearer** [reply]
How might someone use the old Heat.exe with WiX v5?

Does the old Heat.exe require a VS Extension or a NuGet package or...?  How would someone get it?  Does it need to come from WiX v3?

**Reply by md-mcl** [reply]
@robmen Arrived here after searching for the same error message - please could you confirm whether this is a paid-only feature?

**Reply by robmen** [reply]
The error message:

> error WIX9100: HeatWave Build Tools license failure. No license found. Checked for licenses at the standard folder or below the directory: '...'. For more information see: https://www.firegiant.com/wix/wep-documentation/getting-started/firegiant-licensing/

comes from `FireGiant.HeatWave.BuildTools.*` nupkgs, which are commercial WiX extensions from FireGiant. They are documented at https://docs.firegiant.com/heatwave/build-tools/

PS: The URL in the error message's redirect was broken. That should be fixed now (you may need to clear your browser cache to get the redirect to update).
